### PR TITLE
Adding "Introduction to Ray" Tutorial

### PR DESCRIPTION
**Pull request by LeonLuttenberger**
### Feature or Bugfix
- Documentation

### Detail
- Adding tutorial 034 for Ray basics

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 5c87a8676363665870b4031a279ff7f0689f638d
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d022288a-cac5-47f4-b14a-97b346053deb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: 5c87a8676363665870b4031a279ff7f0689f638d
* Result: SUCCEEDED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=63a826d6-848f-4bab-9c67-9dfe66d74ef1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: 5c87a8676363665870b4031a279ff7f0689f638d
* Result: SUCCEEDED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=35957c75-0717-4d20-b374-65a525f108f3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: a7828bc1d9be81aced263cb30301e3905797dc44
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=54af7ee8-8eac-4163-8609-4e78f2fbc8a2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: a7828bc1d9be81aced263cb30301e3905797dc44
* Result: SUCCEEDED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b36d5371-580f-42ee-9552-eacb40f3cb3e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: a7828bc1d9be81aced263cb30301e3905797dc44
* Result: SUCCEEDED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=446d5c5d-ad1e-44af-b832-f2c3d59f866e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: tutorials/034 - Distributing Calls using Ray.ipynb [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+    "import awswrangler as wr\n",
+    "print(f\"Execution Engine: {wr.config.execution_engine}\")\n",
+    "print(f\"Memory Format: {wr.config.memory_format}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note: This behaviour can be turned off setting the environment variables `EXECUTION_ENGINE=python` or `MEMORY_FORMAT=pandas`. Alternatively, the commands can be run on a remote Ray cluster. This is further described in the next tutorial section."
```
Comment (kukushking) [inline tutorials/034 - Distributing Calls using Ray.ipynb:75]: Doesn't config automatically expect "WR_" prefix?

File: tutorials/034 - Distributing Calls using Ray.ipynb [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "%pip install \"awswrangler[ray,modin]==3.0.0b2\""
```
Comment (kukushking) [inline tutorials/034 - Distributing Calls using Ray.ipynb:30]: Minor - need to add this file to `.bump2version.cfg` then

File: tutorials/034 - Distributing Calls using Ray.ipynb [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+    }
+   },
+   "source": [
+    "[![AWS SDK for pandas](_static/logo.png \"AWS SDK for pandas\")](https://github.com/aws/aws-sdk-pandas)\n",
+    "\n",
+    "# 34 - Distributing Calls Using Ray\n",
+    "\n",
+    "AWS SDK for pandas supports distribution of specific calls using [ray](https://docs.ray.io/) and [modin](https://modin.readthedocs.io/en/stable/).\n",
+    "\n",
+    "When this is enabled, data loading methods return Modin data frame instead of Pandas data frames. Modin provides seamless integration and compatibility with existing Pandas code, with the benefit of distributing operations across your Ray instance. The benefit is being able to operate at a much higher scale."
```
Comment (malachi-constant) [inline tutorials/034 - Distributing Calls using Ray.ipynb:17]: Possible nitpick, should this be _"Modin data **frames**_ or "**a** Modin data frame". Not clear if singular or plural when reading?

**Review by malachi-constant — APPROVED** [review]
Nice !

File: tutorials/034 - Distributing Calls using Ray.ipynb (hunk 1 above)
Comment (LeonLuttenberger) [reply tutorials/034 - Distributing Calls using Ray.ipynb:75]: Good catch. Thanks

File: tutorials/034 - Distributing Calls using Ray.ipynb (hunk 2 above)
Comment (LeonLuttenberger) [reply tutorials/034 - Distributing Calls using Ray.ipynb:30]: I'm not sure we want to do that. Once we release `3.0.0`, we'll change this to just `pip install awswrangler[ray,modin]`, without needing to specify the version. It's what we do in all our other tutorials.

File: tutorials/034 - Distributing Calls using Ray.ipynb (hunk 3 above)
Comment (LeonLuttenberger) [reply tutorials/034 - Distributing Calls using Ray.ipynb:17]: Good catch

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 33eed9a0a0f9899d8662bd9575fac97b61a87540
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ebfdf5e6-d34a-4019-84e4-fe39aab689c1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: 33eed9a0a0f9899d8662bd9575fac97b61a87540
* Result: SUCCEEDED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5f111337-5ead-467b-b8fb-3863b5cfc283%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: 33eed9a0a0f9899d8662bd9575fac97b61a87540
* Result: SUCCEEDED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=24a2ea94-ab2c-4740-b672-bb575432e9eb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: tutorials/034 - Distributing Calls using Ray.ipynb (hunk 2 above)
Comment (kukushking) [reply tutorials/034 - Distributing Calls using Ray.ipynb:30]: Yeah but until then why not ensure the version is up-to-date? Can't hurt. Also we already do this for 034 tutorial and a bunch of others (for the documentation links mainly)

Comment (LeonLuttenberger) [reply tutorials/034 - Distributing Calls using Ray.ipynb:30]: Fair point.

**Pull request merged**